src/main/java/com/puppycrawl/tools/checkstyle/utils/TypeDeclarationDescription.java

```diff
@@ -0,0 +1,90 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2022 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Maintains information about the type of declaration.
+ * Any ast node of type {@link TokenTypes#CLASS_DEF} or {@link TokenTypes#INTERFACE_DEF}
+ * or {@link TokenTypes#ENUM_DEF} or {@link TokenTypes#ANNOTATION_DEF}
+ * or {@link TokenTypes#RECORD_DEF} is considered as a type declaration.
+ */
+public class TypeDeclarationDescription {
```

Comment:
I see no need for separate class in utils package. This is used only in one check, so it is better to move it there and make it private static inner class.

Reply:
For now, this is used by only this check, I plan to use it in UnusedLocalVariable too, had this committed earlier, I will send a separate fix for it.

Reply:
I get that, but my point is to do not include this-will-be-needed-somewhere-else code until there is an actual need for it.

So I suggest to make extraction to common place with separate commit in next PR for issue you mentioned.

Please keep in mind that actual fix can be changed, and such things "for future" can be useless. Therefore it is better to keep code clean.

Reply:
Got it, done.